### PR TITLE
feat(arena): port combat and first-floor authoritative play

### LINKED ISSUE #136: Stage 4: port Arena combat, AI, collisions and death
## Summary

Port Arena attacks, consumables, enemy behavior, collision/damage/death ordering into the same Runtime application, retaining the frozen Unity-only game as the reference.

## Why

Stage 4 of #129 follows inventory migration #134 / #135. Combat must preserve f32 arithmetic and iteration order, including damage queued during an update, reverse projectile/grenade traversal and player death taking priority over clear/rewards.

## Scope

- Weapon A/B, cooldowns, blades, normal/heavy projectiles, medkits and targeted grenades.
- Enemy movement, attacks and boss behavior rules; stable entity IDs and ordered damage/death events.
- Swept projectile hits/walls, grenade flight/explosions, transient effects, shield clock integration and immutable death result values.
- First-floor entry/clear/death path in the migration scene so combat can be exercised through normal controls. Endless progression, repeating boss rewards and final default-scene switch remain stage 5.
- Frozen stock-input comparison through the first combat floor, plus targeted reference cases for simultaneous death, hit ordering, shields, consuming edges and attack release gates.

## Out of Scope

Tanu, TSQ1 replay UI, CLI/FFI and content-authoring integration. Do not add public network commands for invented damage or authoritative state patches.

## Definition of Done

- [x] The same first-floor inputs produce the same HP, kills, transforms, timers, IDs and phase changes as the preserved reference.
- [x] Damage/death and transient event ordering follow the fixed update contract, including simultaneous player/last-enemy death.
- [x] Consumption resolves once per press at a gameplay tick, clears on pause/disconnect, and respects valid grenade aim and full-HP medkit retention.
- [x] Unity renders authoritative enemies, projectiles and effects and can play the migrated combat slice.
- [x] Rust/Unity validation, CI and review fixes pass and the PR is merged into develop.

## Verification

Dev Container fmt/clippy/test/doc; baseline integrity and shared recorded-input comparisons; edge-case combat fixtures/tests; real Unity scene/network tests. Keep targeted proof distinct from the full 11F/21F progression gate in stage 5.

## Expected Files or Areas

apps/demo-game Arena state/combat rules, comparison harness, Unity input/projection frontend, reference fixture tooling and contract/evidence docs.

## Notes for Codex

Begin implementation from merged develop after #135. Preserve the C# reference arithmetic/processing order and original baseline files. Do not weaken comparisons to accommodate a different hit/death result; float tolerance is 1e-4 absolute only for continuous state.

Completed in merged PR #137. Dev Container: 142 tests/doctests and fmt/Clippy/docs passed. Unity: 47 EditMode / 13 PlayMode with live host. All CI checks passed; command lifecycle event review finding fixed and resolved. Evidence: doc/verification/arena-combat/results.json.

### PR DESCRIPTION
**Pull request by Nagitch**
## Problem and behavior

Endless Arena's Kitu path had inventory and movement but no playable combat. This change runs weapons, consumables, enemy AI, swept hits, queued damage/death and transient presentation in the authoritative 60 Hz application. Unity submits controls and renders the complete projection; the migration scene now plays preparation through the first floor's clear or death.

Combat preserves the frozen C# update order and f32 arithmetic. Simultaneous last-enemy/player death records the kill but enters results before any clear/reward. Z/X intents coalesce per tick, require release after UI changes, clear on pause and never consume again when a message is retried. Full endless progression, boss rewards and the default-scene switch remain stage 5.

## Validation

- Dev Container: fmt, all-target/all-feature Clippy, 142 Rust tests/doctests, and API documentation passed.
- Frozen C# input prefix: 470 ticks, 43 complete state checkpoints and 16 command outcomes match. Object field sets, actor order, IDs, HP and deaths are exact; continuous fields use absolute tolerance 1e-4.
- Combat regressions cover simultaneous death/event order, blade cooldown/cone, closest swept target/overlap/walls/range, grenade aim/delay, medkit ordering and boss tell/recovery. Runtime tests cover coalescing, pause clearing and duplicates.
- Unity 6000.6.0f1: 47 EditMode and 13 PlayMode tests passed, with the real WebSocket host enabled. The actual migration scene exercises virtual Input System press gating, grenade consumption/paused flight, projected enemies and a first-floor clear.
- Baseline source and frozen fixture hashes remain verified. Evidence: `doc/verification/arena-combat/results.json`.

The C# baseline contains state checkpoints and command receipts, not a raw domain-event trace; explicit event assertions are documented separately. Batch tests do not claim screenshots or a native player build. The demo app remains internal (`publish=false`); no publication attempted.

Closes #136. Stage 4 of #129.

Review fix: successful start/menu lifecycle transitions now emit ordered phase events before their command receipts. Rejected, duplicate and no-change requests do not emit extra transitions; the new regression passes.

### CODE REVIEW

File: apps/demo-game/src/arena/combat.rs [hunk 1]
```diff
@@ -0,0 +1,1095 @@
...
+    fn phase_to(&mut self, phase: i32, tick: i64, output: &mut OscBundle) {
+        let previous = self.phase;
+        self.phase = phase;
+        self.portal_available = phase == 1 || phase == 4;
+        self.chest_available = phase == 1 || (phase == 4 && self.floor % 5 == 0);
+        emit(
+            output,
+            tick,
+            "/game/arena/phase",
+            json!({"previous":previous,"phase":phase,"floor":self.floor,
```
[the remark this answers is not in the document]
Comment (Nagitch) [reply apps/demo-game/src/arena/combat.rs]: Fixed in the latest commit: successful start/menu transitions emit the same ordered phase event before their command receipt. Rejected and duplicate requests do not emit events, and menu while already in Opening does not invent a transition. A Runtime regression verifies 0→1→0→1 order, source tick/output order and receipt placement; all 142 Rust tests/doctests and fmt/Clippy/docs pass. The frozen full-state comparisons remain unchanged.
